### PR TITLE
Small tidy up for the Tools page

### PR DESCRIPTION
**Pull request by janfaracik**
**Before and after**
image

Some very small tweaks to the Tools page to make it a little nicer
* Removed the border at the top of the page to be consistent with other pages
* Section headers are no longer duplicated

If there's any issues with these changes do let me know (e.g. if a plugin is doing something completely custom and these changes don't suit). If all is okay, ~~I'll update the Git plugin to be inline too.~~

Git plugin is updated and released.

### Testing done

* The page should work and perform exactly as before, just with a few UI tweaks 

### Proposed changelog entries

- Small tidy up for the Tools page

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7716"></a>

### CODE REVIEW

File: core/src/main/resources/lib/form/rowSet.jelly [hunk 1]
```diff
@@ -35,12 +35,6 @@ THE SOFTWARE.
     <st:attribute name="ref">
       id of the thing that serves as the group head, if that's available separately
     </st:attribute>
-    <st:attribute name="isFormItem">
```
Comment (janfaracik) [inline core/src/main/resources/lib/form/rowSet.jelly:38]: This attribute isn't being used, so removed it.

https://github.com/search?q=org%3Ajenkinsci%20isFormItem%3D&type=code

File: core/src/main/resources/lib/form/rowSet.jelly [hunk 2]
```diff
@@ -35,12 +35,6 @@ THE SOFTWARE.
...
-      Removes the "jenkins-form-item" class if false
-    </st:attribute>
-    <st:attribute name="class">
```
Comment (janfaracik) [inline core/src/main/resources/lib/form/rowSet.jelly:41]: Nor is class 

https://github.com/search?q=org%3Ajenkinsci+%3Arowset&type=code

**Review by timja — COMMENTED** [review]
~Add installer is broken:~
~button doesn't work and there's a random dropdown element next to it.~
~see e.g. git or jdk plugin on tools page~

Edit: cached javascript from other branch

----

rest looks good apart from git plugin needing updating

**Comment by janfaracik** [comment]
> button doesn't work and there's a random dropdown element next to it. see e.g. git or jdk plugin on tools page

image

Seems to be okay on my end, unless I'm in the wrong spot?

**Comment by timja** [comment]
Ah it was cached javascript from https://github.com/jenkinsci/jenkins/pull/7718 (the prototype js change that's in there that breaks a bunch of stuff.

Looks good apart from the git plugin needing updating then

**Comment by janfaracik** [comment]
> Looks good apart from the git plugin needing updating then

Awesome, I'll update that. For xssPrevented_usingToolInstallation_withJustDisplayName, is the test still necessary as the section labels/descriptions have been removed?

**Comment by timja** [comment]
I think its not needed anymore, but not the easiest to tell.

**Comment by daniel-beck** [comment]
> Just the one way to add a tool now (verses two before)

Since you're basically reverting [JENKINS-43581](https://issues.jenkins-ci.org/browse/JENKINS-43581), please explain why that original change was unjustified / unnecessary. While it's ugly, it is quite convenient.

**Comment by janfaracik** [comment]
> > Just the one way to add a tool now (verses two before)
> 
> Since you're basically reverting [JENKINS-43581](https://issues.jenkins-ci.org/browse/JENKINS-43581), please explain why that original change was unjustified / unnecessary. While it's ugly, it is quite convenient.

That's understandable and I understand the convenience. Happy to restore it.

--

I spent some time coming up with a concept for this page for the future - would you mind checking it out and letting me know what you all think?

https://jenkins-redesign.vercel.app/settings/tools

The main changes are: 
* One button to handle adding tools (at the very top of the page, no more scrolling)
* Tools are still grouped, but are now all collapsible by their section header
* Tools aren't expanded by default, you'd have to click them to open (how valuable is having all the tools editable at once? I'm not sure)

**Comment by daniel-beck** [comment]
> One button to handle adding tools (at the very top of the page, no more scrolling)

I expect this is doable, but require quite a bit of work. The most difficult part might be having a succinct summary for each tool how it's installed. Note that each tool can have any number of installers that are applicable to different label expressions. While it might not be used all that often, it's what we have and need to accommodate. If you're happy with "12 installers" or similar as label there, seems reasonable.

(And FTR I am very critical around making different parts of Jenkins look and work wildly differently from each other rather than consistently -- which was BTW my major negative feedback when the second Add button was originally added. Your concept goes _way_ further.)

**Comment by janfaracik** [comment]
Any thoughts on whether xssPrevented_usingToolInstallation_withJustDisplayName is still needed or not @daniel-beck?

File: core/src/main/resources/hudson/tools/ToolInstallation/global.jelly [hunk 3]
```diff
@@ -23,25 +23,21 @@ THE SOFTWARE.
 -->
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form" xmlns:local="local">
-  <f:section title="${descriptor.displayName}">
+  <f:section title="${%title(descriptor.displayName)}">
     <d:taglib uri="local">
       <d:tag name="section">
         <j:set var="escapeEntryTitleAndDescription" value="true"/>
```
Comment (daniel-beck) [inline core/src/main/resources/hudson/tools/ToolInstallation/global.jelly:29]:
```suggestion
        <!-- While there's no longer a title/description for f:entry at the top level, keep this around for f:entries in tool config pages to be safe -->
        <j:set var="escapeEntryTitleAndDescription" value="true"/>
```

**Comment by daniel-beck** [comment]
> Any thoughts on whether xssPrevented_usingToolInstallation_withJustDisplayName is still needed

It's safe to remove. It confirmed that unsafe tool display names were safe due to how `f:entry` handles `title` attributes. No `title` attribute, no need for the test. Besides, regular users cannot control tool display names, so this was always an improvement rather than a fix.

**Review by timja — APPROVED** [review]
LGTM

**Review by daniel-beck — APPROVED** [review]

**Pull request merged**